File: screener/criteria/plugins/intraday_breakout.py
"""Stocks breaking through 52w high intraday on volume surge."""

from __future__ import annotations

from tradingview_screener import col

from screener.criteria import criterion


@criterion("intraday_breakout")
def intraday_breakout() -> list:
    return [
        col("close").above_pct("price_52_week_high", 0.97),
        col("relative_volume_10d_calc") >= 2.0,
        col("change") >= 1.5,
        col("EMA5") > col("EMA20"),
    ]

File: screener-rs/src/screeners/criteria/intraday_breakout.rs
use super::{CmpOp, Predicate, cmp, field, val};

pub const NAME: &str = "intraday_breakout";

pub fn predicates() -> Vec<Predicate> {
    vec![
        Predicate::AbovePct {
            field: "close",
            other: "price_52_week_high",
            pct: 0.97,
        },
        cmp(field("relative_volume_10d_calc"), CmpOp::Gte, val(2.0)),
        cmp(field("change"), CmpOp::Gte, val(1.5)),
        cmp(field("EMA5"), CmpOp::Gt, field("EMA20")),
    ]
}
